### ISSUE TITLE
Add Feature: More keyboard shortcuts: c for archive card

### ISSUE THREAD

**Issue by xet7** [wekan/wekan#1878]
In Trello, there is:
- moving from card to another with arrow keys
- when your mouse cursor is above card, or with arrow keys you have moved to card, and you press "l" (small L), then labels popup opens.
- also, when card is open, and you press "l", then labels popup open
- also other functionality should work with keyboard.
- etc https://trello.com/shortcuts

Add to Wekan:
- [x] c for archive card
- [ ] l for labels popup

**Comment by hatl** [comment]
Would also be greate if you could add "c" for "archive card"

**Comment by Kiemes** [comment]
Also numbers 1,2,3 for adding green, red, orange labels to cards would be great.

**Comment by JonasDralle** [comment]
I would definitely like to press 'c' to archive a card

**Comment by alexanderdd** [comment]
I feel "a" is much better to signify "archive" than "c"! (also for other languages than English)
How are people supposed to remember c for archive?

off topic: I also don't understand the logic behind many other shortcuts.. "q" for "show my cards" - why not "m"? "w" for "toggle board sidebar" - why not "b" or "s"?

**Comment by xet7** [comment]
@alexanderdd 

The shortcut logic is from Trello:
https://trello.com/shortcuts

Wekan has some features that are similar like those at Trello and Jira, and some that those do not have. Wekan does not have any code/images/etc from Trello and Jira.

**Comment by gerroon** [comment]
Is Ctrl-c/x supported in Wekan if so is it browser speficif?Because it does not work onunder FF 71

**Comment by xet7** [comment]
@gerroon 

Do you mean Snap or Docker? Docker version has this change already. I will make new release today to get this change to Snap.

This feature works like this:
1. Click card to open it.
2. Press `c` to archive card.

FF 71 has bugs that prevent some features of Wekan working, I need to check later are they fixed in FF 72.

**Comment by gerroon** [comment]
@xet7 I do not mean `c` I mean `Ctrl-c, Ctrl-x, Ctrl-v` to copy paste cards, it was referred in the trello shortcuts and it was mentioned that Wekan follows the Trello's mapping.

thanks

**Comment by xet7** [comment]
@gerroon 

Current Wekan shortcuts are listed at Shortcuts popup, that link is at Wekan board sidebar bottom right corner. AFAIK only those are implemented. Those `Ctrl-c, Ctrl-x, Ctrl-v` are not implemented currently.

I was only replying [above comment](https://github.com/wekan/wekan/issues/1878#issuecomment-562871250) about why `c` is for archive card.

I can implement more shortcuts, if someone uses my [Commercial Support](https://wekan.team/commercial-support/) to pay something for my development time.

Alternatively, some Wekan contributor can develop more shortcuts and submit pull request to master branch.

**Comment by xet7** [comment]
Also, there is a lot of differences between Wekan and Trello, I have not checked are all existing shortcuts the same as in Trello.

**Comment by ClaudiaK21** [comment]
Referring to @gerroon 
We would also like a shortcut for **copy/pasting cards**.  if the ctrl + C / X  do not exist, is there a another way to easily do it?
_(I have tried to press/hold a card to copy and move it while "plus" pressing ctrl or alt or shift, but it does not work this way too)_

**Comment by xet7** [comment]
@ClaudiaK21 

If you mean copying cards that have checklist:
1. Click card
2. Click card hamburger
3. Click `Copy Checklist Template to many cards`

**Comment by ClaudiaK21** [comment]
No, I meant the whole card, a faster way to copy & paste instead of 
Click card
Click card hamburger
Copy card
choose list
etc

Some users like us, we have a lot of boards and lists and cards, it would make life easier ;-)

**Comment by ClaudiaK21** [comment]
> Current Wekan shortcuts are listed at Shortcuts popup, that link is at Wekan board sidebar bottom right corner

When having a lot of entries in the activity area, it is nearly very hard to "remember" where the wekan shortcut list link is (or even a lot users might not know about it at all, because you have to scroll down a long list until the very end.

Is it possible to give it another position at the top instead of at the bottom?
and the Link picture is very small at the moment
![grafik](https://user-images.githubusercontent.com/79307821/113713314-789b7d80-96e7-11eb-9f99-7f8106d2c3ef.png)

**Comment by mohammadZahedian** [comment]
> Also numbers 1,2,3 for adding green, red, orange labels to cards would be great.

numbers for label would be great and also "d" for duedate...
they will save hours in month.

**Comment by mohammadZahedian** [comment]
> > Also numbers 1,2,3 for adding green, red, orange labels to cards would be great.
> 
> numbers for label would be great and also "d" for duedate... they will save hours...

**Comment by mohammadZahedian** [comment]
> In Trello, there is:
> 
> * moving from card to another with arrow keys
> * when your mouse cursor is above card, or with arrow keys you have moved to card, and you press "l" (small L), then labels popup opens.
> * also, when card is open, and you press "l", then labels popup open
> * also other functionality should work with keyboard.
> * etc https://trello.com/shortcuts
> 
> Add to Wekan:
> 
> * [x]  c for archive card
> * [ ]  l for labels popup
is it possible to use "d" for duedate ?
archive
labels
duedate
are actions that users use frequently...
but due date is not in the checklist.

**Comment by mohammadZahedian** [comment]
> In Trello, there is:
> 
> * moving from card to another with arrow keys
> * when your mouse cursor is above card, or with arrow keys you have moved to card, and you press "l" (small L), then labels popup opens.
> * also, when card is open, and you press "l", then labels popup open
> * also other functionality should work with keyboard.
> * etc https://trello.com/shortcuts
> 
> Add to Wekan:
> 
> * [x]  c for archive card
> * [ ]  l for labels popup

is it possible to use "d" for duedate ?
archive
labels
duedate
are actions that users use frequently...
but due date is not in the checklist.

**Comment by mohammadZahedian** [comment]
@xet7 
"d" for dueate
it is very necessary

**Comment by xet7** [comment]
@mohammadZahedian 

I tried to add "d" for duedate, but did not get it working yet.

**Comment by mohammadZahedian** [comment]
> @mohammadZahedian
> 
> I tried to add "d" for duedate, but did not get it working yet.

@xet7 any improvment?

**Comment by xet7** [comment]
@mohammadZahedian 

I have many deadlines for features, fixes and upgrades at paid Commercial Support. Non-paid features and fixes do not have deadlines, they come when I have some free time, if possible.

**Comment by xet7** [comment]
@mohammadZahedian 

If someone is in hurry to get some feature or fix, they can submit PR.

**Comment by hatl** [comment]
@xet7 could you please bring back the "c to archive" shortcut?
i actually paid some money for that :wink:
also: ` and % are not the easiest one's to reach on a german keyboard

**Comment by xet7** [comment]
@hatl 

As you see from this discussion https://github.com/wekan/wekan/commit/ba0fdaef72393632ca80b42a3c5d2ee5f5e0c76e#commitcomment-150376465 , problem is, that having "c to archive" shortcut breaks Ctrl + C when copying text from card-details view (@NadavTasher
means, that copying text moves the card to archive, which is why @NadavTasher originally moved it to '-'), at See https://github.com/wekan/wekan/pull/5576

**Comment by xet7** [comment]
That is also why I commented there, that then it would be better to not have any shortcuts.

Any ideas welcome.

**Comment by xet7** [comment]
Also, there will be later a lot of changes to UI, because of accessibility and design improvements.

**Comment by NadavTasher** [comment]
Can make the shortcut (Only the ones that have a system action in combination with Ctrl) do nothing when card-details is open

**Comment by xet7** [comment]
@NadavTasher 

> Can make the shortcut (Only the ones that have a system action in combination with Ctrl) do nothing when card-details is open

Shortcuts should be same as in Trello https://trello.com/shortcuts , if shortcuts are enabled. But default should be to not have shortcuts enabled, I think. Currently I don't know how many know about and use shortcuts. I currently do not use shortcuts.

**Comment by NadavTasher** [comment]
All of my colleagues use the shortcuts, It's their favorite feature.
Having 'c' as a shortcut will re-introduce the bug - I think changing it is necessairy even if it breaks the Trello similarity.

**Comment by mimZD** [comment]
@xet7 
Can't we give the ability for the admin to set these in the settings?

With all these different opinions, it's hard to come to a conclusion. I think it should be left to the admins to decide what shortcuts to use based on their organization.
We suggest something basic and they have the ability to change the shortcuts however they want.

**Comment by hatl** [comment]
I'm no JavaScript expert - but shouldn't some code like the following be able to distinguish C from CTRL-C?

```js
Mousetrap.bind('c', evt => {
  // Check if the Ctrl key is pressed
  if (evt.ctrlKey) {
    console.log('CTRL-C pressed');
    // Perform the action for CTRL-C
  } else {
    console.log('C pressed');
    const cardId = getSelectedCardId();
    if (!cardId) {
      return;
    }
    // Perform the action for C
  }
});
```
